services/checks/checks.go

```diff
@@ -565,46 +574,83 @@ type sttCheckResult struct {
 	result    check.Result
 }
 
-func (s *Service) processResults(ctx context.Context, check check.Check, target target, resID string) ([]sttCheckResult, error) {
+// StarlarkScriptData represents the data we need to pass to the binary to run starlark scripts.
+type StarlarkScriptData struct {
+	CheckName    string                   `json:"name"`
+	Script       string                   `json:"script"`
+	CheckVersion uint32                   `json:"version"`
+	ScriptInput  []map[string]interface{} `json:"input"`
+}
+
+func (s *Service) processResults(ctx context.Context, sttCheck check.Check, target target, resID string) ([]sttCheckResult, error) {
 	nCtx, cancel := context.WithTimeout(ctx, resultTimeout)
 	r, err := s.waitForResult(nCtx, resID)
 	cancel()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get action result")
 	}
 
-	funcs, err := getFuncsForVersion(check.Version)
+	l := s.l.WithFields(logrus.Fields{
+		"name":       sttCheck.Name,
+		"id":         resID,
+		"service_id": target.serviceID,
+	})
+
+	input := &StarlarkScriptData{
+		CheckName:    sttCheck.Name,
+		Script:       sttCheck.Script,
+		CheckVersion: sttCheck.Version,
+		ScriptInput:  r,
+	}
+
+	cmdCtx, cancel := context.WithTimeout(ctx, scriptTimeout)
+	defer cancel()
+
+	process := exec.CommandContext(cmdCtx, "pmm-managed-starlark")
+	pdeathsig.Set(process, syscall.SIGTERM)
+	err = unix.Setrlimit(unix.RLIMIT_CPU, &unix.Rlimit{Cur: cpuLimit, Max: cpuLimit})
+	if err != nil {
+		l.Warn("Failed to limit CPU usage: ", err)
+	}
+	//err = unix.Setrlimit(unix.RLIMIT_DATA, &unix.Rlimit{Cur: memoryLimit, Max: memoryLimit})
```

Comment:
@AlekSi if I set a memory limit then I get an `out of memory` error, I am guessing the reason for that is that the memory is being set for the parent process too rather than just the child process. Maybe I am doing it the wrong way, can you take a look at this and tell me what I am doing wrong?

Reply:
Please uncomment this code to log warning, I will take a look

Reply:
done

Reply:
As I understand yes, `Setrlimit` adds resource limitations to the current process and then it is inherited by child processes. 
https://man7.org/linux/man-pages/man2/setrlimit.2.html
```
       A child process created via fork(2) inherits its parent's resource
       limits.  Resource limits are preserved across execve(2).
```

Reply:
> an unprivileged process may set only **its** soft limit to a value in the range from 0 up to the hard limit, and (irreversibly) lower its hard limit.

Seems that solution is move limitations setting into the child process.

Reply:
> Seems that solution is move limitations setting into the child process.

yeah but I can't find a good way to do that, there is one way, that would be to get the child's pid and set the limits against the pid, but by the the pid the process would have already started.

Reply:
I could run a shell instance and run ulimit along with the binary something like this `process := exec.CommandContext(cmdCtx, "/bin/sh", "-c", "ulimit -m 100000000; pmm-managed-starlark")`

This will return `exit status 137` on running scripts like these 
```python
script: |
          def check(rows):
              return [1] * (1 << 30-1)
```

cc @AlekSi

Reply:
I could run a shell instance and run ulimit along with the binary something like this process := exec.CommandContext(cmdCtx, "/bin/sh", "-c", "ulimit -m 100000000; pmm-managed-starlark")

This will return exit status 137 on running scripts like these

script: |
          def check(rows):
              return [1] * (1 << 30-1)

cc @AlekSi

Reply:
I think we should just set limits in `pmm-managed-starlark` binary itself